### PR TITLE
GH-1034: Add deprecated version of IndexType::IndexTypes()

### LINKED ISSUE #1034: API backwards compatibility: IndexType::IndexTypes() return type changed
The return type of IndexType::IndexTypes() changed. It now is a ``const std::vector<IntrusivePtr<Type>>&;`` before it was a ``const type_list*``.

This will beak basically any code using it since the API changed a bit. I assume this one we probably don't want to/can't easily fix - it might be worth documenting it somewhere.

### PR DESCRIPTION
**Pull request by timwoj**
There were deprecated versions added for most methods where just the return value changed during the conversion to IntrusivePtr. Apparently this one was missed.

Fixes #1034

### CODE REVIEW

File: src/Type.h [hunk 1]
```diff
@@ -457,10 +457,19 @@ class IndexType : public Type {
...
 
-	[[deprecated("Remove in v4.1.  Use GetIndices().")]]
+	[[deprecated("Remove in v4.1. Use GetIndices().")]]
 	TypeList* Indices() const		{ return indices.get(); }
 
-	const std::vector<IntrusivePtr<Type>>& IndexTypes() const
+	[[deprecated("Remove in v4.1. Use GetIndexTypes().")]]
+	const type_list* IndexTypes() const
+		{
+		type_list* tl = new type_list();
```
Comment (jsiwek) [inline src/Type.h:466]: This method didn't used to return an allocated object for the caller to have to clean up, so I think what we'd actually need to do for backward compat is change `TypeList` to store a duplicate/mirrored `type_list` member.

Probably would want to do a sanity check of what impact the duplicate has on mem. usage too, but wouldn't guess Type size matters much there.

**Comment by jsiwek** [comment]
Thanks for checking into this, also forgot to mention this breakage was noted in `NEWS`:

https://github.com/zeek/zeek/blob/eb1a408b6aa3bb0f53d07be097ba6ad77fe4a846/NEWS#L103-L104

No real reason other than didn't think at the time of storing a duplicate member like I suggested, but can also remove that `NEWS` entry if we find a way to add in deprecated versions of the old methods.

**Comment by timwoj** [comment]
Ah, didn't realize that this was in the NEWS already (and didn't think to check).

**Comment by timwoj** [comment]
Local testing says that the amount of extra memory used keeping an extra set of attributes and types is negligible.

branch:
```
27.29 116664
25.14 117508
25.08 116120
31.05 116872
32.96 116336
avg: 116700
```

master:
```
26.72 115284
31.50 116212
35.58 114952
31.74 115504
30.24 118892
avg: 116168.8
```

**Pull request merged**